symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/signal/OboSignalService.java

```diff
@@ -0,0 +1,169 @@
+package com.symphony.bdk.core.service.signal;
+
+import com.symphony.bdk.gen.api.model.BaseSignal;
+import com.symphony.bdk.gen.api.model.ChannelSubscriber;
+import com.symphony.bdk.gen.api.model.ChannelSubscriptionResponse;
+import com.symphony.bdk.gen.api.model.Signal;
+
+import org.apiguardian.api.API;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Service interface exposing OBO-enabled endpoints to manage signals information.
+ */
+@API(status = API.Status.STABLE)
+public interface OboSignalService {
+
+  /**
+   * Lists signals on behalf of the user.
+   * {@link SignalService#listSignals(Integer, Integer)}
+   *
+   * @param skip  The number of signals to skip.
+   * @param limit Maximum number of signals to return. Default is 50, maximum value is 500.
+   * @return List of signals that the user has created and public signals to which they have subscribed.
+   * @see <a href="https://developers.symphony.com/restapi/reference#list-signals">List Signals</a>
+   */
+  List<Signal> listSignals(@Nullable Integer skip, @Nullable Integer limit);
+
+  /**
+   * Lists signals on behalf of the user with default limit equal 50.
+   * {@link SignalService#listSignals()}
+   *
+   * @return List of signals that the user has created and public signals to which they have subscribed.
+   * @see <a href="https://developers.symphony.com/restapi/reference#list-signals">List Signals</a>
+   */
+  List<Signal> listSignals();
+
+  /**
+   * Lists paginated stream of signals on behalf of the user.
+   * {@link SignalService#listSignalsStream(Integer, Integer)}
+   *
+   * @param chunkSize size of elements to retrieve in one call. Optional and defaults to 50.
+   * @param totalSize Total maximum number of signals to return. Optional and defaults to 50.
+   * @return a {@link Stream} containing the signals.
+   * @see <a href="https://developers.symphony.com/restapi/reference#list-signals">List Signals</a>
+   */
+  Stream<Signal> listSignalsStream(@Nullable Integer chunkSize, @Nullable Integer totalSize);
```

Comment:
Adding @nullable is good idea, we could do that for existing methods too

Reply:
(like message service)

Reply:
I think we can do it in another PR to not make the difference of this PR become huge 😄

Reply:
Could be done as part of https://perzoinc.atlassian.net/browse/APP-3183